File: prdeploy-api/src/PrDeploy.Api.Models/Settings/DeploySettings.cs
namespace PrDeploy.Api.Models.Settings;
public class DeploySettings
{
    //// This has to match prdeploy-webhooks/src/models/settings/repo-settings.ts.
    public string? DeployWorkflow { get; set; }
    public List<EnvironmentSettings>? Environments { get; set; } = new ();
    public List<ServiceSettings>? Services { get; set; } = new();
    public string? DefaultEnvironment { get; set; }
    public string? ReleaseEnvironment { get; set; }
    public JiraSettings? Jira { get; set; }
    public BuildsSettings? Builds { get; set; }
    public SlackSettings? Slack { get; set; }
    public string? PrdeployPortalUrl { get; set; }
    public BadgeSettings? Badge { get; set; }
}

File: prdeploy-webhooks/src/models/settings/deploy-settings.ts
import { Lifecycle, scoped } from 'tsyringe';
import { EnvironmentSettings } from './environments-settings';
import { SlackSettings } from './slack-settings';
import { BuildsSettings } from './builds-settings';
import { BadgeSettings } from './badge-settings';
import { ServiceSettings } from './service-settings';
import { JiraSettings } from './jira-settings';

@scoped(Lifecycle.ContainerScoped)
export class DeploySettings {
  owner: string;
  repo: string;
  deployWorkflow: string;
  environments: EnvironmentSettings[];
  services: ServiceSettings[];
  defaultEnvironment: string;
  releaseEnvironment: string;
  defaultBranch: string;
  jira: JiraSettings;
  builds: BuildsSettings;
  slack: SlackSettings;
  prdeployPortalUrl: string;
  badge: BadgeSettings;
}
